### PR TITLE
Example

### PR DESCRIPTION
**Pull request by MartinKliemank**
Added an example folder and an example file which simulates TGV3D and uses several reporters and does some plotting ect. to demonstrate general usage of lettuce, with hopefully enough comments for easy understanding

@dominikwilde your suggestions sadly didn't reach my branch for some reason, I tried to include what i had noted / remembered, please remind me if stuff is missing

### CODE REVIEW

**Comment by Olllom** [comment]
Good initiative, thanks!
Why don't you add the example as a jupyter notebook?
They are perfectly suited for what you are doing here and github renders them nicely.

We could then also include them in the [docs](https://lettuceboltzmann.readthedocs.io/en/latest/). (Either directly through a Sphinx add-on or via a link)

**Comment by dominikwilde** [comment]
The idea of adding a jupyter notebook is nice and helpful.

I realized that my suggestions went into a separate branch "[patch-1"](https://github.com/dominikwilde/lettuce/blob/patch-1/examples/generalUsage.py), for which I missed to create a pull request. But I think most of my remarks are in the code.

**Comment by Olllom** [comment]
> The idea of adding a jupyter notebook is nice and helpful.
> 

Any updates on this?

**Comment by MartinKliemank** [comment]
I'm sorry, I spoke to Dominik about it in person but didn't update you here. We do like the idea and also thought it would be good to also add an example which goes a bit more into detail about the inner workings. 
Sadly I'm currently preparing for exams, also the primary objective of my Masterprojekt (masters project I guess?) is to continue Dominik's enstrophy comparison of the flow around a 2D and 3D building/object, so I put work on the example aside for now.

**Comment by Olllom** [comment]
Alright. Thanks for the update. I will assume that the example is correct and merge it for now.

File: lettuce/io.py [hunk 1]
```diff
@@ -48,9 +48,14 @@ def __call__(self, i, t, f):
...
             u = self.flow.units.convert_velocity_to_pu(self.lattice.u(f))
             p = self.flow.units.convert_density_lu_to_pressure_pu(self.lattice.rho(f))
-            self.point_dict["p"] = self.lattice.convert_to_numpy(p[0, ..., None])
-            for d in range(self.lattice.D):
-                self.point_dict[f"u{'xyz'[d]}"] = self.lattice.convert_to_numpy(u[d, ..., None])
+            if self.lattice.D == 2:
+                self.point_dict["p"] = self.lattice.convert_to_numpy(p[0, ..., None])
+                for d in range(self.lattice.D):
+                    self.point_dict[f"u{'xyz'[d]}"] = self.lattice.convert_to_numpy(u[d, ..., None])
+            else:
```
Comment (Olllom) [inline lettuce/io.py:55]: Wait ... is this the correct behavior for 1D and 3D?

File: lettuce/io.py [hunk 2]
```diff
@@ -48,9 +48,14 @@ def __call__(self, i, t, f):
...
+                self.point_dict["p"] = self.lattice.convert_to_numpy(p[0, ...])
+                for d in range(self.lattice.D):
+                    self.point_dict[f"u{'xyz'[d]}"] = self.lattice.convert_to_numpy(u[d, ...])
```
Comment (Olllom) [inline lettuce/io.py:58]: Can you remind me: Why do we need 2 separate cases here? Does pyevtk require a different shape for 2D and 3D?

Comment (McBs) [reply lettuce/io.py:58]: Yes it does. See https://github.com/Olllom/lettuce/pull/14

**Comment by dominikwilde** [comment]
@Olllom 
I think the VTK issue is solved and already in the master branch.

I retested the example file and it works. From my point of view it can be merged.

**Pull request merged**